sql/src/main/java/org/opensearch/sql/sql/domain/SQLQueryRequest.java

```diff
@@ -77,29 +76,35 @@ public SQLQueryRequest(JSONObject jsonContent, String query, String path,
     this.params = params;
     this.format = getFormat(params);
     this.sanitize = shouldSanitize(params);
-    // TODO hack
-    this.cursor = cursor == null ? "" : cursor;
+    this.cursor = cursor;
   }
 
   /**
    * Pre-check if the request can be supported by meeting ALL the following criteria:
    *  1.Only supported fields present in request body, ex. "filter" and "cursor" are not supported
    *  2.Response format is default or can be supported.
    *
-   * @return  true if supported.
+   * @return true if supported.
    */
   public boolean isSupported() {
-    return (isCursor() || isOnlySupportedFieldInPayload())
-        && isSupportedFormat();
+    var noCursor = !isCursor();
```

Comment:
Should these be booleans instead of var?

Reply:
`var` could be anything, even `boolean` as here. My favorite lazy developer feature.